aws-solutions/aws-instance-scheduler
# Use regional STS endpoint to assume cross-account roles

Acquire STS token from the regional endpoint of the region the lambda is currently executing in rather than the global endpoint. This causes STS to acquire a V2 token that is compatible with all regions instead of just standard regions.

manual test setup:
Hub stack deployed in US-East-1
Spoke stack deploy in Us-East 1

EC2 deployed Instances in spoke stack:
us-east-1 (same region): schedulable 
us-west-2 (cross-region, standard): schedulable
me-south-1 (cross-region, opt-in): schedulable
eu-central-2 (cross-region, boto filter error): not schedulable

this last one (eu-central-2) will be addressed in a separate PR covering the boto3 describe_regions filtration story

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
